### DISCUSSION TITLE
"Mute Myself" choice should be persistent across sessions

### DISCUSSION THREAD

**Discussion by yurivict** [jamulussoftware/jamulus#1318] [Ideas]
When the user chooses to be muted he later assumes that he is muted and it is safe to talk. This choice is currently lost after restart, and there is likelihood that this user would inadvertently cause disruption with his sound while thinking that it is muted.

**Comment by dcorson-ticino-com** [comment]
I disagree, if someone is not muted they will hear themselves and note it immediately.
If however they start a session and are muted they may not notice, or remember if the last session was a while ago.

**Reply by npostavs** [reply]
> If however they start a session and are muted they may not notice

Maybe this could be solved by blinking or something if sound is detected while muted.

**Reply by softins** [reply]
This would become very annoying, as some people mute themselves so that they can play along without disturbing the group (for whatever reason). There is already a prominent red notice displayed above the mixer panel when Mute Myself is on.
![image](https://user-images.githubusercontent.com/3224952/113270858-d2f4a280-92d1-11eb-8cbb-37da59209d3a.png)

**Comment by yurivict** [comment]
> If however they start a session and are muted they may not notice, or remember if the last session was a while ago.

Then make it persistent for a while, with a reasonable timeout.

**Comment by yurivict** [comment]
A special Settings checkbox can be added: "Persistent Self-Mute", so users would be able to choose for themselves.

**Comment by ghost** [comment]
Users should have to unmute themselves by default, and whenever Jamulus decides to mute them. This way people will get trained to check that they are unmuted. No need to save mute status because this would default to muted.

**Comment by yurivict** [comment]
Or, again, support all possible behaviors and make user decide what suits him best.

Add a combobox to settings with choices like "Mute Self when started", "Unmute Self when started", "Blink Unmute Self when sound is detected", etc.

There is no need to find a one-fits-all solution when many opposing opinions are present. Having the user to choose the behavior gets past devs opinion differences.

**Reply by npostavs** [reply]
Adding more options makes things more complicated. In this case, I don't think the added complexity is worth it. Even though I'd prefer muting state be saved, I'd still vote for the status quo over adding multiple ways of configuring it.

**Reply by yurivict** [reply]
I don't know why do people complain when some listener joins with random noises on.
